### DISCUSSION TITLE
Add class on body with background addon?

### DISCUSSION THREAD

**Discussion by apfelbox** [storybookjs/storybook#16176] [Help]
Hi,

so I want to have changeable backgrounds on my components. In my SCSS these background changes are base classes and my components respond to that.

Now, is it possible to have the background addon set a class on the body? Because if just the color changes, I have no way to react in my component to that.

Thanks

**Comment by certainlyakey** [comment]
~Same question.~

Found an answer. In `preview.ts`:

```
decorators: [
    ...
    (story, context) => {
      const bgNameDefault = backgroundsValues[0].name;
      const bgName = context.parameters.backgrounds.default || bgNameDefault;
      if (bgName === bgNameDefault) {
        document.documentElement.classList.remove('is-color-scheme-dark');
        document.documentElement.classList.add('is-color-scheme-light');
      } else {
        document.documentElement.classList.toggle('is-color-scheme-light');
        document.documentElement.classList.add('is-color-scheme-dark');
      }
      return story();
    },
  ],
```

where `backgroundsValues` is a regular array of values that configures the Backgrounds addon (`{name: string, value: string}[]`).
